### DISCUSSION TITLE
FR: Option to force backup on the db/restore console command

### DISCUSSION THREAD

**Discussion by lukasNo1** [craftcms/cms#14534] [Ideas] [answered]
Hey, we are running an automatic deployment where we have a db rollback option which essentially just calls
craft db/restore.

Now the user gets asked in an interactive shell if he wants to create a backup first.
An option here would be nice (like --backup) that would always run a backup, without asking.

Essentially i don't want anyone running the restore step without making a backup first. Right now my solution is to just run the db/backup command before the rollback.

https://github.com/craftcms/cms/blob/5.0.0-beta.5/src/console/controllers/DbController.php#L275

**Comment by brandonkelly** [comment]
Not sure I’m following. Who is the user? How is `db/restore` getting initiated? Ideally you should only be giving CLI access to people who you can absolutely trust to do the right thing, given all the damage that can be done with it.

**Comment by lukasNo1** [comment]
The `db/restore`  is initiated in a shell script we can execute to push a specific db state  (mostly used to update the staging db)

The users are the other devs that can deploy the project. So everyone should know to make a backup :)
I mainly just want to make sure nobody forgets to do it (especially cause “no” is the default answer)

**Reply by brandonkelly** [reply, marked as answer]
Gotcha. Calling `db/backup` before `db/restore` via the shell script would be the recommended way of automating this.
